lib/ansible/module_utils/dimensiondata.py

```diff
@@ -0,0 +1,393 @@
+    Get the list of available regions whose vendor is Dimension Data.
+    """
+    check_libcloud_or_fail()
+
+    # Get endpoints
+    all_regions = API_ENDPOINTS.keys()
+
+    # filter to only DimensionData endpoints
+    region_list = filter(lambda i: i.startswith('dd-'), all_regions)
+
+    # Strip prefix
+    regions = [region[3:] for region in region_list]
+
+    return regions
+
+
+def get_network_domain_by_name(driver, name, location):
+    """
+    Get a network domain object by its name
+    """
+    networks = driver.ex_list_network_domains(location=location)
+    found_networks = filter(lambda x: x.name == name, networks)
+
+    if not found_networks:
+        raise UnknownNetworkError("Network '%s' could not be found" % name)
+
+    return found_networks[0]
+
+
+def get_network_domain(driver, locator, location):
+    """
+    Get a network domain object by its name or id
+    """
+    if is_uuid(locator):
+        net_id = locator
+    else:
+        name = locator
+        networks = driver.ex_list_network_domains(location=location)
+        found_networks = filter(lambda x: x.name == name, networks)
+
+        if not found_networks:
+            raise UnknownNetworkError("Network '%s' could not be found" % name)
+
+        net_id = found_networks[0].id
+
+    return driver.ex_get_network_domain(net_id)
+
+
+def get_vlan(driver, locator, location, network_domain):
+    """
+    Get a VLAN object by its name or id
+    """
+    if is_uuid(locator):
+        vlan_id = locator
+    else:
+        vlans = driver.ex_list_vlans(location=location,
+                                     network_domain=network_domain)
+        found_vlans = filter(lambda x: x.name == locator, vlans)
+
+        if not found_vlans:
+            raise UnknownVLANError("VLAN '%s' could not be found" % locator)
+
+        vlan_id = found_vlans[0].id
+
+    return driver.ex_get_vlan(vlan_id)
+
+
+def get_mcp_version(driver, location):
+    """
+    Get a locations MCP version
+    """
+    # Get location to determine if MCP 1.0 or 2.0
+    location = driver.ex_get_location_by_id(location)
+    if 'MCP 2.0' in location.name:
+        return '2.0'
+    return '1.0'
+
+
+def is_uuid(u, version=4):
+    """
+    Test if valid v4 UUID
+    """
+    try:
+        uuid_obj = UUID(u, version=version)
+        return str(uuid_obj) == u
+    except:
+        return False
+
+
+def expand_ip_block(block):
+    """
+    Expand public IP block to show all addresses
+    """
+    addresses = []
+    ip_r = block.base_ip.split('.')
+    last_quad = int(ip_r[3])
+    address_root = "%s.%s.%s." % (ip_r[0], ip_r[1], ip_r[2])
+    for i in range(int(block.size)):
+        addresses.append(address_root + str(last_quad + i))
+    return addresses
+
+
+def get_public_ip_block(module, driver, network_domain, block_id=False,
+                        base_ip=False):
+    """
+    Get public IP block details
+    """
+    # Block ID given, try to use it.
+    if block_id is not 'False':
```

Comment:
I don't think this works?  block_id defaults to the boolean False in the parameters to this function.  So you probably need to check:
``` python
if block_id is not False:
```

rather than checking against a string.